### PR TITLE
add web-ext config and ignore files from build, fixes #135

### LINKED ISSUE #135: Audit files in built web-ext
The README says you can build the code via <kbd>web-ext build</kbd>. After trying to build the add-on (from the root directory), it appears that the add-on is ~2MB and includes all the source .svelte files and package.json and a lot of unnecessary things. Not sure if we should only be building from the ./public/ directory or if I'm doing something wrong...

```sh
npx web-ext build --overwrite-dest 
Applying config file: ./package.json
Building web extension from /Volumes/Dev/github/mozilla/ion-core-addon
Your web extension is ready: /Volumes/Dev/github/mozilla/ion-core-addon/web-ext-artifacts/ion_core-2.0.zip


ls -lash web-ext-artifacts/ 
4000 -rw-r--r--@  1 pdehaan  staff   2.0M  4 Nov 15:28 ion_core-2.0.zip


zipinfo -1 web-ext-artifacts/ion_core-2.0.zip | grep -Ev "/$" | sort
CODE_OF_CONDUCT.md
LICENSE
README.md
core-addon/DataCollection.js
core-addon/FirefoxPrivilegedApi.js
core-addon/FirefoxPrivilegedApi.schema.json
core-addon/IonCore.js
core-addon/Storage.js
core-addon/background.js
core-addon/content-script.js
docs/ION_CORE_PUBLISHING.md
docs/STUDY_PUBLISHING.md
docs/index.html
images/ion.svg
manifest.json
package-lock.json
package.json
public/addon-build/background.js
public/addon-build/content-script.js
public/build/bundle.css
public/build/bundle.css.map
public/build/bundle.js
public/build/bundle.js.map
public/common.css
public/favicon.png
public/favicon.svg
public/fonts/Inter-Bold.woff
public/fonts/Inter-Bold.woff2
public/fonts/Inter-BoldItalic.woff
public/fonts/Inter-BoldItalic.woff2
public/fonts/Inter-Italic.woff
public/fonts/Inter-Italic.woff2
public/fonts/Inter-Regular.woff
public/fonts/Inter-Regular.woff2
public/fonts/Metropolis-Bold.woff
public/fonts/Metropolis-Bold.woff2
public/fonts/Metropolis-Medium.woff
public/fonts/Metropolis-Medium.woff2
public/fonts/ZillaSlab-Bold.woff
public/fonts/ZillaSlab-Bold.woff2
public/fonts/ZillaSlab-BoldItalic.woff
public/fonts/ZillaSlab-BoldItalic.woff2
public/fonts/ZillaSlab-Regular.woff
public/fonts/ZillaSlab-Regular.woff2
public/fonts/ZillaSlab-RegularItalic.woff
public/fonts/ZillaSlab-RegularItalic.woff2
public/global.css
public/index.html
public/locally-available-studies.json
public/rally.css
public/tokens.css
rollup.config.addon.js
rollup.config.js
scripts/setupTypeScript.js
src/App.svelte
src/EnrollmentButton.svelte
src/StudyCard.svelte
src/api/firefox.js
src/api/web-extension.js
src/api/web.js
src/components/Accordion.svelte
src/components/Arrow00.svelte
src/components/Arrow01.svelte
src/components/Arrow02.svelte
src/components/Button.svelte
src/components/Layout.svelte
src/components/Main.svelte
src/components/Modal.svelte
src/components/MozillaRallyLogo.svelte
src/components/OnboardingCTAContainer.svelte
src/components/forwardEvents.js
src/components/icons/Carat.svelte
src/components/icons/CaretRight.svelte
src/components/icons/ExternalLink.svelte
src/copy/EnrollmentDisclaimer.svelte
src/copy/HowItWorks.svelte
src/copy/ValueProposition.svelte
src/copy/WhyItMatters.svelte
src/create-store.js
src/main.js
src/mocks/available-studies.js
src/mocks/firefox-mock.js
src/regions/StudyList.svelte
src/routes/demographics/CallToAction.svelte
src/routes/demographics/Content.svelte
src/routes/terms-of-service/CallToAction.svelte
src/routes/terms-of-service/Content.svelte
src/routes/welcome/Content.svelte
src/routes/welcome/FirstRunArrow.svelte
src/routes/welcome/Graph.svelte
src/routes/welcome/Profile.svelte
src/routes/welcome/Study.svelte
src/stores.js
stories/flow/DemographicsView.svelte
stories/flow/OnboardingFlowView.svelte
stories/flow/TermsView.svelte
stories/flow/WelcomeView.svelte
stories/flow/flow.stories.js
stories/protocol/ButtonView01.svelte
stories/protocol/button.stories.js
tests/core-addon/DataCollection.test.js
tests/core-addon/IonCore.test.js
tests/core-addon/Storage.test.js
tests/core-addon/hooks.js
```

### PR DESCRIPTION
**Pull request by rhelmer**
I don't see a great way to include vs. exclude files, unfortunately :( this does the job for right now.

### CODE REVIEW

**Comment by pdehaan** [comment]
Current list of files in the built add-on:

```sh
npm run build-addon
npx web-ext build
zipinfo -1 web-ext-artifacts/ion_core-2.0.zip | sort | grep -Ev "/$"

core-addon/DataCollection.js
core-addon/FirefoxPrivilegedApi.js
core-addon/FirefoxPrivilegedApi.schema.json
core-addon/IonCore.js
core-addon/Storage.js
core-addon/background.js
core-addon/content-script.js
images/ion.svg
manifest.json
public/addon-build/background.js
public/addon-build/content-script.js
public/build/bundle.css
public/build/bundle.css.map
public/build/bundle.js
public/build/bundle.js.map
public/common.css
public/favicon.png
public/favicon.svg
public/fonts/Inter-Bold.woff
public/fonts/Inter-Bold.woff2
public/fonts/Inter-BoldItalic.woff
public/fonts/Inter-BoldItalic.woff2
public/fonts/Inter-Italic.woff
public/fonts/Inter-Italic.woff2
public/fonts/Inter-Regular.woff
public/fonts/Inter-Regular.woff2
public/fonts/Metropolis-Bold.woff
public/fonts/Metropolis-Bold.woff2
public/fonts/Metropolis-Medium.woff
public/fonts/Metropolis-Medium.woff2
public/fonts/ZillaSlab-Bold.woff
public/fonts/ZillaSlab-Bold.woff2
public/fonts/ZillaSlab-BoldItalic.woff
public/fonts/ZillaSlab-BoldItalic.woff2
public/fonts/ZillaSlab-Regular.woff
public/fonts/ZillaSlab-Regular.woff2
public/fonts/ZillaSlab-RegularItalic.woff
public/fonts/ZillaSlab-RegularItalic.woff2
public/global.css
public/index.html
public/locally-available-studies.json
public/rally.css
public/tokens.css
```

Re: > _"I don't see a great way to include vs. exclude files, unfortunately..."_

Not sure if moving the add-on files into an addon specific subdirectory would help. Possibly like this:

```sh
addon/
├── core-addon/...
├── images/ion.svg
├── manifest.json
└── public/...
```

Then, when we run <kbd>web-ext build -s addon</kbd> (or better yet, edit the web-ext-config.js config to specify the `sourceDir: "addon"` so it works with all commands), it should only include files in that specific directory and we can probably lose all the files listed in the `ignoreFiles[]` array in the web-ext config.

File: web-ext-config.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+module.exports = {
```
Comment (pdehaan) [inline web-ext-config.js:1]: Needs MPL header. ❤️ 

```
/home/circleci/project/web-ext-config.js
  1:1  error  Missing notice header  notice/notice

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

File: web-ext-config.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  verbose: true,
+  // Command options:
+  build: {
+    overwriteDest: true,
+  },
+  run: {
+    firefox: "nightly",
+    pref: ["extensions.experiments.enabled=true"],
+  },
+  ignoreFiles: [
```
Comment (Dexterp37) [inline web-ext-config.js:12]: I think we should also add `core-addon/**/*` and `public/**/*.map`: the manifest includes the bundled code which lives in `public/addon-build`, the original code is not required in the packaged addon.

**Review by Dexterp37 — CHANGES_REQUESTED** [review]

**Comment by rhelmer** [comment]
> Re: > _"I don't see a great way to include vs. exclude files, unfortunately..."_
> 
> Not sure if moving the add-on files into an addon specific subdirectory would help. Possibly like this:
> 
> ```shell
> addon/
> ├── core-addon/...
> ├── images/ion.svg
> ├── manifest.json
> └── public/...
> ```
> 
> Then, when we run web-ext build -s addon (or better yet, edit the web-ext-config.js config to specify the `sourceDir: "addon"` so it works with all commands), it should only include files in that specific directory and we can probably lose all the files listed in the `ignoreFiles[]` array in the web-ext config.

@Dexterp37 any thoughts on @pdehaan's suggestion above? I'd rather do an "include" vs. "exclude" strategy, this would get us closer to that.

**Comment by Dexterp37** [comment]
> @Dexterp37 any thoughts on @pdehaan's suggestion above? I'd rather do an "include" vs. "exclude" strategy, this would get us closer to that.

I love the idea of rationalizing this repo a bit more! +1 on that (include strategy).

**Review by Dexterp37 — APPROVED** [review]

**Comment by rhelmer** [comment]
> 
> 
> > @Dexterp37 any thoughts on @pdehaan's suggestion above? I'd rather do an "include" vs. "exclude" strategy, this would get us closer to that.
> 
> I love the idea of rationalizing this repo a bit more! +1 on that (include strategy).

Filed issue #209 so we don't lose track of this, I think the PR as it stands is very simple and is better than what we have now so merging :)

**Comment by pdehaan** [comment]
Minor Q (which I can file separately upon request), but I noticed if I build the add-on locally using `npx web-ext build`, and then inspect the bundled files, I see some odd stuff (highlighted in red below) in the "support/" directory:

```diff
zipinfo -1 ./web-ext-artifacts/ion_core-2.0.zip | sort | grep -Ev "/$"

images/ion.svg
manifest.json
public/addon-build/background.js
public/addon-build/content-script.js
public/build/bundle.css
public/build/bundle.js
public/build/protocol-extra.css
public/build/protocol.css
public/common.css
public/default-study-icon.png
public/favicon.png
public/favicon.svg
public/fonts/Inter-Bold.woff
public/fonts/Inter-Bold.woff2
public/fonts/Inter-BoldItalic.woff
public/fonts/Inter-BoldItalic.woff2
public/fonts/Inter-Italic.woff
public/fonts/Inter-Italic.woff2
public/fonts/Inter-Regular.woff
public/fonts/Inter-Regular.woff2
public/fonts/Metropolis-Bold.woff
public/fonts/Metropolis-Bold.woff2
public/fonts/Metropolis-Medium.woff
public/fonts/Metropolis-Medium.woff2
public/fonts/ZillaSlab-Bold.woff
public/fonts/ZillaSlab-Bold.woff2
public/fonts/ZillaSlab-BoldItalic.woff
public/fonts/ZillaSlab-BoldItalic.woff2
public/fonts/ZillaSlab-Regular.woff
public/fonts/ZillaSlab-Regular.woff2
public/fonts/ZillaSlab-RegularItalic.woff
public/fonts/ZillaSlab-RegularItalic.woff2
public/global.css
public/index.html
public/locally-available-studies.json
public/rally.css
public/texture.png
public/tokens.css
-support/README.md          <----- THIS
-support/package-lock.json  <----- THIS
-support/package.json       <----- THIS
support/rally.js
```

---

TL:DR: Do we need to include the following files in the support/ directory in the bundled add-on:
- support/README.md
- support/package-lock.json
- support/package.json

They're only about 850B unzipped, so the file size isn't a concern.

**Pull request merged**